roles/external_access/tasks/create_external_access.yaml

```diff
@@ -66,17 +66,32 @@
   retries: "{{ external_access_resource_wait_retries }}"
   delay: "{{ external_access_resource_wait_delay }}"
 
-- name: Set external_access_worker_node_ip
+- name: Get esi node name of an agent
+  ansible.builtin.set_fact:
+    external_access_worker_node_name: >-
+      {{
+      agents.resources[0].spec.hostname
+      }}
+
+- name: Get network details from esi
+  ansible.builtin.command: >-
+    openstack esi node network show {{ external_access_worker_node_name }} -f json
```

Comment:
Not something that needs to be fixed (especially since I think this is all eventually going to be refactored to use a bare metal service), but I think `openstack esi node network list --node MOC-R4PAC24U35-S3A --network network-larstest2` might have gotten what you wanted!

Reply:
I've made this change; I guess it simplifies the expression required to extract the node ip.